src/shared/format-table.c

```diff
@@ -387,6 +388,15 @@ int table_add_cell_full(
         if (!data)
                 type = TABLE_EMPTY;
 
+        if (type == TABLE_STRV) {
+                buf = strv_join(data, "\n");
+                if (!buf)
+                        return -ENOMEM;
+
+                data = buf;
+                type = TABLE_STRING;
+        }
```

Comment:
adding this data type makes a ton of sense, but I think we should store this properly as char**, and not convert implicitly into a string. We expose table_get() and table_get_at() after all which are supposed to return the original data in some form, but if we don't have the data that's not going to work :-(

Also, for the JSON generation It would make a lot of sense to generate a array of strings for this case.